docs/scos/dev/upgradability-services/upgradability-reference.md

```diff
@@ -110,3 +110,231 @@ The method name should be unique to the extent of making it impossible to accide
 2. Override the core method with the custom one you've created in the previous step.
 
 When the core method is overriden with a custom one, re-evaluate the code. The same error shouldn't be returned.
+
+
+
+## Private API class was extended or used
+
+Private API updates can break backward compatibility. So, backward compatibility in minor releases is not guaranteed in the private API. For example, you use a core class or method on the project level. If it is updated or removed, it might change behavior or cause issues.
+
+The following core classes are exceptions, and you can use and extend them on the project level:
+
+* All the classes from the modules:
+  * Kernel
+  * Bootstrap
+  * Development
+
+* Particular classes:
+  * Facade
+  * Factory
+  * Entity manager
+  * Repository
+  * Dependency provider
+  * Config
+  * ConfigurationProvider
+
+
+
+### Using custom classes on the project level
+
+To avoid unexpected issues and achieve the same result, replace core classes with custom ones.
+
+### Example of code that can cause upgradability errors
+
+`CustomerAccessForm` extends from `Spryker\Zed\CustomerAccessGui\Communication\Form\CustomerAccessForm` class from the core level.
+
+```php
+<?php
+...
+**
+ * @method \Spryker\Zed\CustomerAccessGui\Communication\CustomerAccessGuiCommunicationFactory getFactory()
+ */
+class CustomerAccessForm extends SprykerCustomerAccessForm
+{
+    public const OPTION_CONTENT_TYPE_ACCESS_MANAGEABLE = 'OPTION_CONTENT_TYPE_ACCESS_MANAGEABLE';
+    public const OPTION_CONTENT_TYPE_ACCESS_NON_MANAGEABLE = 'OPTION_CONTENT_TYPE_ACCESS_NON_MANAGEABLE';
+    public const OPTION_CONTENT_TYPE_ACCESS_NON_MANAGEABLE_DATA = 'OPTION_CONTENT_TYPE_ACCESS_NON_MANAGEABLE_DATA';
+...
+}
+```
+
+Related error in the Evaluator output:
+```text
+------------------------------------------------------------------------------------------------------------------------
```

Comment:
I'd suggest making the dotted line in this example and the examples below shorter as it makes the code sample bulky on the Storefront